### ISSUE TITLE
[BUG] Streaming MistralAi leaves out blank chunk

### ISSUE THREAD

**Issue by derflotzi** [langchain4j/langchain4j#1218]
**When streaming chat completion from MistralAI, the blank chunk `" "` is omitted. Usually occurs before numerical values.**

- Current behaviour: "In2020, Germany's inflation rate was0.5%."
- Expected behavior: "In 2020, Germany's inflation rate was 0.5%."

This happens because `DefaultMistralAiClient` uses `isNotNullOrBlank(chunk)` for filtering. Correct would be `isNotNullOrEmpty(chunk)`.

**Code from DefaultMistralAiClient (Version 0.31.0):**
```
// Line 136 - 142
MistralAiChatCompletionChoice choice = chatCompletionResponse.getChoices().get(0);

String chunk = choice.getDelta().getContent();
if (isNotNullOrBlank(chunk)) { // FIXME should be isNotNullOrEmpty(chunk)
    contentBuilder.append(chunk);
    handler.onNext(chunk);
}
```

**To Reproduce**
```
@Test
public void DefaultMistralAiClient_allow_blank() throws InterruptedException, ExecutionException {
    StreamingChatLanguageModel model = MistralAiStreamingChatModel.builder()
            .apiKey(System.getenv("MISTRAL_AI_API_KEY"))
            .modelName(MistralAiChatModelName.MISTRAL_SMALL_LATEST)
            .temperature(0d)
            .build();

    List<String> chunks = new ArrayList<>();
    CompletableFuture<AiMessage> future = new CompletableFuture<>();

    String userMessage = "What was inflation rate in germany in 2020? Answer in 1 short sentence. Begin your answer with 'In 2020, ...'";
    model.generate(userMessage, new StreamingResponseHandler<>() {
        @Override
        public void onNext(String token) {
            chunks.add(token);
        }

        @Override
        public void onComplete(Response<AiMessage> response) {
            System.out.println(chunks.stream().map(s -> "'" + s + "'").collect(Collectors.joining(" ")));
            future.complete(response.content());
        }

        @Override
        public void onError(Throwable error) {
            future.completeExceptionally(error);
        }
    });

    // results in: "In2020, Germany's inflation rate was0.5%."
    assertThat(future.get().text()).containsIgnoringCase("In 2020");
}
```

- LangChain4j version: 0.31.0
- LLM(s) used: MistralAI (DefaultMistralAiClient)
- Java version: 21.0.3

**Comment by langchain4j** [comment]
@derflotzi thanks a lot for reporting! Could you please open a PR?

cc @czelabueno

**Comment by czelabueno** [comment]
@derflotzi good catch! Let us know if you can open a PR to fix it, otherwise we can do it.

**Comment by derflotzi** [comment]
Could you please create the PR?

**Comment by langchain4j** [comment]
Fixed by #1243
